R/carehomes.R

```diff
@@ -817,10 +796,207 @@ carehomes_severity <- function(p) {
+##'   from `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_G_D_CHR A list with `date` and `value` for changes in p_G_D (the
+##'   probability of individuals requiring hospitalisation dying in a care home)
+##'   for care home residents. If `value` is `NULL` then the value for the
+##'   oldest age group is used. If `value` is scalar then `date` can be `NULL`
+##'   or unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_star A list with `date` and `value` for changes in p_star (the
+##'   probability of patients being confirmed as covid on admission to
+##'   hospital). If `value` is `NULL` then the value
+##'   from `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @return A list of severity parameters
+##'
+##' @export
+carehomes_parameters_severity <- function(dt,
+                                          severity = NULL,
+                                          p_C = list(date = NULL,
+                                                     value = NULL),
+                                          p_H = list(date = NULL,
+                                                     value = NULL),
+                                          p_H_CHR = list(date = NULL,
+                                                         value = NULL),
+                                          p_ICU = list(date = NULL,
+                                                       value = NULL),
+                                          p_H_D = list(date = NULL,
+                                                       value = NULL),
+                                          p_ICU_D = list(date = NULL,
+                                                         value = NULL),
+                                          p_W_D = list(date = NULL,
+                                                       value = NULL),
+                                          p_G_D = list(date = NULL,
+                                                       value = NULL),
+                                          p_G_D_CHR = list(date = NULL,
+                                                           value = NULL),
+                                          p_star = list(date = NULL,
+                                                        value = NULL)) {
+
   severity <- sircovid_parameters_severity(severity)
   severity <- lapply(severity, carehomes_severity)
-  severity$p_G_D[length(severity$p_G_D)] <- p_death_carehome
+
+  get_p_step <- function(x, name) {
+
+    p_name <- paste0("p_", name)
+    p <- x[[p_name]]
+    p_value <- get(p_name)$value
```

Comment:
Use of `get` is usually a flag you're doing something odd. The trick here is to assemble a list as quickly as possible.